src/core/macro.scala

```diff
@@ -13,6 +13,18 @@ object Macro:
   inline def repeated[T]: List[(String, Boolean)] = ${repeated[T]}
   inline def typeInfo[T]: TypeInfo = ${typeInfo[T]}
   inline def summonOption[T]: Option[T] = ${summonOptionImpl[T]}
+  inline def isProduct[T]: Boolean = ${isProductImpl[T]}
+  inline def isSum[T]: Boolean = ${isSumImpl[T]}
+
+  def isProductImpl[T: Type](using Quotes): Expr[Boolean] =
+    import quotes.reflect.*
+    val typeSymbol = TypeRepr.of[T].typeSymbol
+    Expr(typeSymbol.isClassDef && typeSymbol.flags.is(Flags.Case))
+
+  def isSumImpl[T: Type](using Quotes): Expr[Boolean] =
+    import quotes.reflect.*
+    val typeSymbol = TypeRepr.of[T].typeSymbol
+    Expr(typeSymbol.flags.is(Flags.Sealed))
```

Comment:
that's probably not enough - shouldn't we check if this is a trait as well? You could write `sealed case class` I suppose, even if it doesn't make sense